### PR TITLE
Correctly handle connections that fail security checks (RIPD-1114):

### PR DESCRIPTION
**Pull request by nbougalis**
* Return error code 400 to the peer along with a descriptive message
* Release the slot and decrement IP connection counters.

### CODE REVIEW

**Comment by JoelKatz** [comment]
:+1:

File: src/ripple/overlay/impl/OverlayImpl.cpp [hunk 1]
```diff
@@ -348,6 +371,27 @@ OverlayImpl::makeRedirectResponse (PeerFinder::Slot::ptr const& slot,
     return response;
 }
 
+std::shared_ptr<Writer>
+OverlayImpl::makeErrorResponse (PeerFinder::Slot::ptr const& slot,
+    http_request_type const& request,
+    address_type remote_address,
+    std::string msg)
+{
+    beast::deprecated_http::message m;
```
Comment (vinniefalco) [inline src/ripple/overlay/impl/OverlayImpl.cpp]: Any chance we can use the new message object?

Comment (nbougalis) [inline src/ripple/overlay/impl/OverlayImpl.cpp]: Yep, on it.

**Comment by nbougalis** [comment]
Merged as b5dbd7942f8896367e65cbc8f58e9bfbce81d953

**Pull request closed without merging**